### PR TITLE
feat: Update 'Create Project/Workspace' Designs

### LINKED ISSUE #6670: Update 'Create Project/Workspace' Designs
The old modal view need to be updated with the new design system styles. 

Designs are ready to be implemented here: https://www.figma.com/file/rdBPNzn7Klk6RG1LHUCE5B/Screen-Designs?type=design&node-id=11718-717530&mode=design&t=fad58Y3Js4m8N56x-11

Placement of the modal + scrim color are defined!

### LINKED ISSUE #6671: Update 'Switch Workspace' Flow & Designs
Instead of using the old modal view, we want to use a menu approach, similar to switching projects. 

Designs are ready here: https://www.figma.com/file/rdBPNzn7Klk6RG1LHUCE5B/Screen-Designs?type=design&node-id=11718-717005&mode=design&t=fad58Y3Js4m8N56x-11

### PR DESCRIPTION
**Pull request by rishi-raj-jain**
fix #6670
fix #6671

### CODE REVIEW

**Comment by rishi-raj-jain** [comment]
How changes look as of now: https://github.com/highlight/highlight/issues/6670#issuecomment-1729018567

**Comment by SpennyNDaJets** [comment]
@rishi-raj-jain Looking good! I see this is coming along, but still in progress. Tagging @julian-highlight for some of the questions, and feel free to tag me when ready for a review

**Comment by SpennyNDaJets** [comment]
@rishi-raj-jain Not sure if this is helpful, but you may be able to use the components:
```
import Modal from '@components/Modal/Modal'
import ModalBody from '@components/ModalBody/ModalBody'
```
that have some of the styles already built in

**Comment by rishi-raj-jain** [comment]
@SpennyNDaJets 

Right now I see that the background that comes on /w/1/new is basically white, am I to not redirect to new workspce/project pages anymore?

**Comment by rishi-raj-jain** [comment]
Let me share my current screenshots

**Comment by rishi-raj-jain** [comment]
Screenshot 2023-09-23 at 10 07 22 AM

**Comment by rishi-raj-jain** [comment]
Screenshot 2023-09-23 at 10 07 40 AM

**Comment by rishi-raj-jain** [comment]
# New Project: Modal

Screenshot 2023-09-23 at 10 09 02 AM

**Comment by rishi-raj-jain** [comment]
# New Project: Modal in typed state

Screenshot 2023-09-23 at 10 15 05 AM

**Comment by rishi-raj-jain** [comment]
@SpennyNDaJets @Vadman97 

While I believe the design is all implemented, the expectation of scrim background is not so clear to me. Currently, there are no background elements to the pages /new or /w/[id]/new so how do I  get the background layout elements to appear there as seen in design?

**Comment by rishi-raj-jain** [comment]
Ah got my way around it. Basically added Intercepting Routes via modifying the router.

**Comment by rishi-raj-jain** [comment]
@SpennyNDaJets @Vadman97

This is ready for review.

**Comment by rishi-raj-jain** [comment]
https://github.com/highlight/highlight/assets/46300090/52ef04c6-e50c-4211-bbef-f98ef6e04c0f

**Comment by rishi-raj-jain** [comment]
@SpennyNDaJets @Vadman97

This is ready for review.

**Comment by rishi-raj-jain** [comment]
@SpennyNDaJets @Vadman97

This is ready for review.

File: packages/ui/src/components/icons/IconOutlineX.tsx [hunk 1]
```diff
@@ -1,22 +1,27 @@
 import React from 'react'
 import { IconProps } from './types'
 
-export const IconOutlineX = ({ size = '1em', ...props }: IconProps) => {
+export const IconOutlineX = ({
+	size = '1em',
+	strokeWidth = '4',
```
Comment (SpennyNDaJets) [inline packages/ui/src/components/icons/IconOutlineX.tsx:6]: Are these changes still needed? I think ideally we would revert these changes so that this icon doesn't perform differently than the other icons.

File: packages/ui/src/components/icons/types.ts [hunk 2]
```diff
@@ -2,4 +2,5 @@ import React from 'react'
 
 export type IconProps = React.SVGAttributes<SVGElement> & {
 	size?: string | number
+	strokeWidth?: string | number
```
Comment (SpennyNDaJets) [inline packages/ui/src/components/icons/types.ts:5]: Same here - think it makes sense to revert this, assuming it is not needed, and supported by all our icons?

**Review by SpennyNDaJets — CHANGES_REQUESTED** [review]
There seems to be a bug with getting redirected when trying to create a new workspace.

Steps to repo:
1. Be logged in and viewing a workspace
2. Click in the top right settings dropdown to "Switch workspace"
3. Click create new workspace

Expected behavior: See new modal to create new workspace
Actual behavior: Redirected to sessions page 

https://github.com/highlight/highlight/assets/17744174/c2936e8f-992b-41cc-915f-e501a19575dc

File: packages/ui/src/components/icons/IconOutlineX.tsx (hunk 1 above)
Comment (rishi-raj-jain) [reply packages/ui/src/components/icons/IconOutlineX.tsx:6]: Per the design IconOutlineX shall have strokeWidth of 8. How else can I do the same?

File: packages/ui/src/components/icons/types.ts (hunk 2 above)
Comment (rishi-raj-jain) [reply packages/ui/src/components/icons/types.ts:5]: Per the design IconOutlineX shall have strokeWidth of 8. How else can I do the same?

File: packages/ui/src/components/icons/IconOutlineX.tsx (hunk 1 above)
Comment (SpennyNDaJets) [reply packages/ui/src/components/icons/IconOutlineX.tsx:6]: Good question! I think it should already be matching the styles of the other modal? I don't believe we are using `IconOutlineX`, but sorry if I am missing something

File: packages/ui/src/components/icons/types.ts (hunk 2 above)
Comment (SpennyNDaJets) [reply packages/ui/src/components/icons/types.ts:5]: Keeping the discussion in https://github.com/highlight/highlight/pull/6682/files#r1338665209

**Comment by rishi-raj-jain** [comment]
@SpennyNDaJets 

So in the AppRouter if none of the path matched "/new", it fallback'ed to 

```
<Route
								path="/*"
								element={
									projectId &&
									(isValidProjectId ||
										projectId === DEMO_PROJECT_ID) ? (
										<ProjectRouter />
									) : isLoggedIn ? (
										<ProjectRedirectionRouter />
									) : (
										<AuthRouter />
									)
								}
							/>
```

which would be incorrect, hence the fix.

**Comment by rishi-raj-jain** [comment]
@SpennyNDaJets 

This is ready to review.

**Comment by rishi-raj-jain** [comment]
@SpennyNDaJets @Vadman97

This PR now fixes both https://github.com/highlight/highlight/issues/6670 and https://github.com/highlight/highlight/issues/6671.

**Comment by rishi-raj-jain** [comment]
So for the workspace design this now works and it looks so cool!:

Screenshot 2023-09-28 at 5 29 11 AM

File: frontend/src/pages/NewProject/NewProject.module.css [hunk 3]
```diff
@@ -24,9 +24,16 @@
...
 .button {
+	width: 100%;
+	height: 30px;
 	display: flex;
-	margin-bottom: 0;
-	margin-top: 0;
+	flex-direction: row;
+	align-items: center;
+	max-width: max-content;
+	padding-top: 1px !important;
```
Comment (SpennyNDaJets) [inline frontend/src/pages/NewProject/NewProject.module.css:33]: Not a huge fan of having to use `!important`, but okay with it if necessary

File: frontend/src/components/Header/Header.tsx [hunk 4]
```diff
@@ -154,6 +160,65 @@ export const Header: React.FC<Props> = ({ fullyIntegrated }) => {
 
 	const inProjectOrWorkspace = isLoggedIn && (projectId || workspaceId)
 
+	const { data: workspacesData, loading: workspacesLoading } =
+		useGetWorkspacesQuery()
+
+	const workspaceOptions = (workspacesData?.workspaces || [])
```
Comment (SpennyNDaJets) [inline frontend/src/components/Header/Header.tsx:166]: Wonder if we can do this in a useMemo, to avoid calculating this on ever rerender?

Something like
```
const workspaceOptions = useMemo(() => {
  const currentWorkspaces = workspacesData?.workspaces || []
  const joinableWorkspaces = workspacesData?.joinable_workspaces || []

  return currentWorkspaces.concat(joinableWorkspaces).map((workspace) => ({
			value: workspace.id,
			displayValue: workspace.name,
			id: workspace.id,
		}))
}, [workspacesData?.workspaces, workspacesData?.joinable_workspaces])

// build JSX outside of useMemo
```

**Review by SpennyNDaJets — APPROVED** [review]
Click test looks good! Just a couple code nits, but let me know if it doesn't make sense or shouldn't be down this way

File: frontend/src/pages/NewProject/NewProject.module.css (hunk 3 above)
Comment (rishi-raj-jain) [reply frontend/src/pages/NewProject/NewProject.module.css:33]: @SpennyNDaJets yeah this is required here

**Comment by rishi-raj-jain** [comment]
@SpennyNDaJets done

**Comment by rishi-raj-jain** [comment]
Let me know if this needs anything. Just so it's not missed, this PR aims at fixing both https://github.com/highlight/highlight/issues/6670 and https://github.com/highlight/highlight/issues/6671.

**Review by SpennyNDaJets — APPROVED** [review]
Looks good, will let @julian-highlight have the final design say

**Comment by rishi-raj-jain** [comment]
@SpennyNDaJets you're running the PR locally to test, correct?

**Comment by rishi-raj-jain** [comment]
> Looks good
 
❤️

**Review by julian-highlight — APPROVED** [review]

**Comment by julian-highlight** [comment]
> > Looks good
> 
> ❤️

looking great!

**Comment by rishi-raj-jain** [comment]
The golden day is here. Let's get this merged? :D

File: frontend/src/pages/Landing/Landing.tsx [hunk 5]
```diff
@@ -14,7 +14,6 @@ export const Landing = ({ children }: { children: ReactNode }) => {
 
 	return (
 		<Box
-			backgroundColor="p12"
```
Comment (Vadman97) [inline frontend/src/pages/Landing/Landing.tsx:17]: this change broke the style of the login page. what pages is it necessary for?
image

Comment (Vadman97) [inline frontend/src/pages/Landing/Landing.tsx:17]: @rishi-raj-jain this change broke the style of the login page. what pages is it necessary for?
image

Comment (rishi-raj-jain) [reply frontend/src/pages/Landing/Landing.tsx:17]: Commented on the other pull request.
